opengovsg/FormSG
# refactor: migrate GET /submission to new endpt

## Problem
This is part 1 of #1520. This PR addresses the fact that URL query parameters should not be used to identify a resource. rather, they should be used to identify _properties_ of a resource. Hence, there is a new controller + endpoint set up so that the api endpoint is restful.

## Solution
1. Remove request parameter from `RequestHandler` and shift it to the url isntead. 
2. Ports over existing controller unit tests and changes it so that the parameter is in the url 
3. Ports over existing integration tests and changes it so that the parameter is in the url

## Manual Tests
1. Create a new storage mode form and make it public. Go to the form and submit a few responses. Go to the admin panel and click on data and click on any response. The panel should be populated with the information of the responses

## Notes 
1. Shifted from `/admin/forms` to`${ADMIN_FORMS_PREFIX}` in tests. Maybe this should have been done in a separate PR? Felt that it was small (and related to the current PR) so i did it here but comments welcome
